pytorch_lightning/loops/fit_loop.py

```diff
@@ -63,12 +63,12 @@ def global_step(self, value: int) -> None:
 
     @property
     def total_batch_idx(self) -> int:
-        """Returns the total number of batches already run (across all epochs)"""
+        """Returns the current batch index (across epochs)"""
         return self.epoch_loop.total_batch_idx
 
     @property
     def batch_idx(self) -> int:
-        """Returns the number of batches already run within this epoch"""
+        """Returns the current batch index (within this epoch)"""
         return self.epoch_loop.batch_idx
```

Comment:
Should we remove/deprecate these properties in the fit loop?

They are not exposed in the trainer. Do we expect users to check them by accessing `trainer.fit_loop.*`?

Reply:
yes, they can be removed or deprecated. 
They are simply there to simplify the previous refactors and to avoid long chains of calls through `trainer.fit_loop.some_loop.another_loop.batch_idx`. And also to make them read only when possible.

> With loop customization, the references could not exist anymore

not necessarily as we may want to introduce intermediate base classes that separate Lightning built-in loops from more versatile loops with hooks and interfaces that are more specific than the abstract Loop base class contains.

but cleaner if we can remove them :)

Reply:
👍 
I'll leave them in this PR but we should revisit all in that file eventually